### ISSUE TITLE
Implement a Script for LLM fine-tuning

### ISSUE THREAD

**Issue by grayJiaaoLi** [amosproj/amos2024ss08-cloud-native-llm#98]
## User story
1. As a ML engineer
2. I want / need to implement a script for LLM fine-tuning on Hetzner machine
3. So that the fine-tuning can easily be performed using high performance GPU

## Acceptance criteria
* Implement the Fine-Tuning functions
    * Integrate the selected fine-tuning tools or solutions
    * Monitor the process
        * GPU usage
        * GPU storage, catch exceptions
        * Errors
* Hyperparameters are systematically evaluated  using tools
* For the evaluation the metric specified in https://github.com/amosproj/amos2024ss08-cloud-native-llm/issues/81 is used
* The results of https://github.com/amosproj/amos2024ss08-cloud-native-llm/issues/92 are incorporated
* (Enable the _checkpoints_ feature)
    * Prevent bad situations: broken down or lost connection

## Definition of done (DoD)
* Bill of Materials in the planning document has been updated
* All the complex logics have been tested
* All feature branches have been merged and closed
* New feature code has been documented
* Potential new licenses have been checked
* All GitHub Actions are passing
* The requirement.txt is updated

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

**Comment by dominic0df** [comment]
already invested SP05, left SP05

**Comment by dominic0df** [comment]
needs debugging in the execution environment so SP02 left, already invested SP03
